tests/Unit/PostmanCollectionWriterTest.php

```diff
@@ -0,0 +1,319 @@
+                'required' => false,
+                'value' => null,
+            ],
+        ];
+
+        $collection = $this->fakeCollection([$fakeRoute]);
+        $writer = new PostmanCollectionWriter($collection, 'fake.localhost');
+        $collection = json_decode($writer->getCollection(), true);
+
+        $variableData = data_get($collection, 'item.0.item.0.request.url.query');
+
+        $this->assertCount(2, $variableData);
+        $this->assertContains([
+            'key' => 'required',
+            'value' => null,
+            'description' => 'A required param with a null value',
+            'disabled' => false,
+        ], $variableData);
+        $this->assertContains([
+            'key' => 'not_required',
+            'value' => null,
+            'description' => 'A not required param with a null value',
+            'disabled' => true,
+        ], $variableData);
+    }
+
+    /**
+     * @dataProvider provideAuthConfigHeaderData
+     */
+    public function testAuthAutoExcludesHeaderDefinitions(array $authConfig, array $expectedRemovedHeaders)
+    {
+        \Config::set('apidoc.postman', [
+            'auth' => $authConfig,
+        ]);
+
+        $route = $this->fakeRoute('some/path');
+        $route['headers'] = $expectedRemovedHeaders;
+        $collection = $this->fakeCollection([$route], 'Group');
+        $writer = new PostmanCollectionWriter($collection, 'fake.localhost');
+        $collection = json_decode($writer->getCollection(), true);
+
+        foreach ($expectedRemovedHeaders as $key => $value) {
+            $this->assertNotContains(compact('key', 'value'), data_get($collection, 'item.0.item.0.request.header'));
+        }
+    }
+
+    public function provideAuthConfigHeaderData()
+    {
+        yield [
+            ['type' => 'bearer', 'bearer' => ['token' => 'Test']],
+            ['Authorization' => 'Bearer Test'],
+        ];
+
+        yield [
+            ['type' => 'apikey', 'apikey' => ['value' => 'Test', 'key' => 'X-Authorization']],
+            ['X-Authorization' => 'Test'],
+        ];
+    }
+
+    public function testApiKeyAuthIsIgnoredIfExplicitlyNotInHeader()
+    {
+        \Config::set('apidoc.postman', [
+            'auth' => ['type' => 'apikey', 'apikey' => [
+                'value' => 'Test',
+                'key' => 'X-Authorization',
+                'in' => 'notheader',
+            ]],
+        ]);
+
+        $route = $this->fakeRoute('some/path');
+        $route['headers'] = ['X-Authorization' => 'Test'];
+        $collection = $this->fakeCollection([$route], 'Group');
+        $writer = new PostmanCollectionWriter($collection, 'fake.localhost');
+        $collection = json_decode($writer->getCollection(), true);
+
+        $this->assertContains([
+            'key' => 'X-Authorization',
+            'value' => 'Test',
+        ], data_get($collection, 'item.0.item.0.request.header'));
+    }
+
+    protected function fakeRoute($path, $title = '')
```

Comment:
Also rename this to make it clearer.  At first I thought it faked an Illuminate Route object. I think `fakeRouteData` would be better.